Userland/Libraries/LibWeb/PixelUnits.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2023, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/PixelUnits.h>
+
+namespace Web {
+
+static int const layout_unit_fractional_bits = 6;
+static constexpr int fixed_point_denominator = 1 << layout_unit_fractional_bits;
+
+CSSPixels::CSSPixels(int value)
+{
+    m_value = value * fixed_point_denominator;
+}
+
+CSSPixels::CSSPixels(unsigned int value)
+{
+    m_value = value * fixed_point_denominator;
+}
+
+CSSPixels::CSSPixels(float value)
```

Comment:
Does this (and the overload for `double`) handle infinity correctly? I'd think we need to set the underlying value to `INT_MAX / fixed_point_denominator` or something.

Reply:
true! and UBSAN also reported this problem so I fixed that by introducing `AK::clamp_to_int(value)` that transforms input value to valid int range.